couchbase/read_preference.hxx

```diff
@@ -0,0 +1,27 @@
+
+/* -*- Mode: C++; tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+/*
+ *   Copyright 2020-Present Couchbase, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+#pragma once
+namespace couchbase
+{
+enum class read_preference {
+    no_preference,
+    selected_server_group,
```

Comment:
Should we add a codedoc here (or maybe in the respective option setters) about where the selected server group can be configured? i.e.`cluster_options::network()::preferred_server_group()` (or `server_group_affinity()` going off the discussion in the RFC?)

Reply:
I will add docs with examples.